packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationBilling.tsx

```diff
@@ -1,24 +1,69 @@
 import { useOrganization } from '@clerk/shared/react';
 
+import { ChangePlanButton, ManagePlanScreen, OrganizationPlanCard } from '../../common/Billing';
+import { Box, descriptors, Spinner } from '../../customizables';
 import { useFetch } from '../../hooks';
 
 export const OrganizationBilling = () => {
   const { organization } = useOrganization();
+  const { data: availablePlans, isLoading: isLoadingAvailablePlans } = useFetch(
+    organization?.getAvailablePlans,
+    'availablePlans',
+  );
+  const { data: currentPlan, isLoading: isLoadingCurrentPlan } = useFetch(organization?.getCurrentPlan, 'currentPlan');
 
-  const { data, isLoading } = useFetch(organization?.getAvailablePlans, {});
-  console.log(data, isLoading);
-
-  if (isLoading) {
-    return <div>Loading</div>;
+  if (isLoadingAvailablePlans || isLoadingCurrentPlan) {
+    return (
+      <Box
+        sx={{
+          width: '100%',
+          height: '100%',
+          position: 'relative',
+        }}
+      >
+        <Box
+          sx={{
+            margin: 'auto',
+            position: 'absolute',
+            left: '50%',
+            top: '50%',
+            transform: 'translateY(-50%) translateX(-50%)',
+          }}
+        >
+          <Spinner
+            size='sm'
+            colorScheme='primary'
+            elementDescriptor={descriptors.spinner}
+          />
+        </Box>
+      </Box>
+    );
   }
 
-  if ((data?.data.length || 0) < 1) {
-    return <>No data</>;
-  }
+  const handleChangePlan = async (planKey: string) => {
+    try {
+      const response = await organization?.changePlan({ planKey });
+      //TODO Handle action after success response
+      console.log(response);
+    } catch (e) {
+      //TODO Show error message if exists
+      console.log(e);
+    }
+  };
 
-  const plan = data?.data.map(({ name, id }) => {
-    return <li key={id}>{name}</li>;
+  const plans = availablePlans?.data.map(({ name, id, features, priceInCents, key }) => {
+    return (
+      <OrganizationPlanCard
+        isCurrentPlan={currentPlan?.key === key}
+        planKey={key}
+        key={id}
+        name={name}
+        features={features}
+        priceInCents={priceInCents / 100}
+        changePlanAction={<ChangePlanButton action={() => handleChangePlan(key)} />}
+      />
+    );
```

Comment:
Based on the assumption that we will not have different cards between users and orgs, can't we move this logic inside `ManagePlaceScreen` ?

I'd either keep the same API but `plans` should be of type "availablePlan", or simple accept children
```tsx
<MangePlanScreen>
  {plans}
</MangePlanScreen>

```

Reply:
Initially like this, but the `changePlanAction` will be different for orgs and users. I moved this outside in order to avoid prop drilling of the  `handleChangePlan` function. 

I can change the `ManagePlanScreen` to accept children instead of plans.